### PR TITLE
Added Mathematical Algorithms 

### PR DESCRIPTION
**Pull request by yashagarwal1999**
Added Mathematical Algorithms like Calculating Catalan Numbers, Euler Toient Function and Fast power (a^b)%c in logn time

### CODE REVIEW

File: Algorithms/Mathematical/catalan_numbers.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#include<bits/stdc++.h>
+#define MOD 1000000007
+#define MOD9 1000000009
+#define forr(i,p,n) for(ll i=p;i<n;i++)
+#define endl "\n"
+typedef long long ll;
+using namespace std;
+
```
Comment (Bhavnaharitsa) [inline Algorithms/Mathematical/catalan_numbers.cpp:22]: Write down sample inputs and outputs in comments

File: Algorithms/Mathematical/eluer-totient.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        result -= result / n;
+    return result;
+}
+
+
+int main(){
+    ios::sync_with_stdio(0);
+    cin.tie(0);cout.tie(0);
+ll n;
+cin>>n;
```
Comment (Bhavnaharitsa) [inline Algorithms/Mathematical/eluer-totient.cpp]: Same

File: Algorithms/Mathematical/fast_power.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    ios::sync_with_stdio(0);
+    cin.tie(0);cout.tie(0);
+ ll a,b,c;
+ cin>>a>>b>>c;
+ ll ans=fast_power(a,b,c);
+ cout<<ans<<endl;
+ 
+
+
+
```
Comment (Bhavnaharitsa) [inline Algorithms/Mathematical/fast_power.cpp]: Mention the algorithm name, it's purpose and sample input-output for others to understand

File: Algorithms/Mathematical/catalan_numbers.cpp (hunk 1 above)
Comment (yashagarwal1999) [reply Algorithms/Mathematical/catalan_numbers.cpp:22]: Added Sample input and output

File: Algorithms/Mathematical/catalan_numbers.cpp [hunk 4]
```diff
@@ -31,13 +33,23 @@ ll n;
...
+Sample Input:
+5
+Sample Output:
+42
+
+Explanation:
+The Input takes an integer N and the output is the Nth Catalan Number.
+Catalan Numbers are (2nCn)/(n+1) where C is thee binomial coefficient.
+These are special types of numbers which denotes the pattern in many competitive coding contests.
+These numbers were found by Belgian mathematician Catalan.
```
Comment (Bhavnaharitsa) [inline Algorithms/Mathematical/catalan_numbers.cpp]: push this explanation on top

File: Algorithms/Mathematical/catalan_numbers.cpp [hunk 5]
```diff
@@ -31,13 +33,23 @@ ll n;
 cin>>n;
```
Comment (Bhavnaharitsa) [inline Algorithms/Mathematical/catalan_numbers.cpp]: no single variables

File: Algorithms/Mathematical/catalan_numbers.cpp [hunk 6]
```diff
@@ -31,13 +33,23 @@ ll n;
...
 ll x=catalan_number(n);
```
Comment (Bhavnaharitsa) [inline Algorithms/Mathematical/catalan_numbers.cpp]: same

File: Algorithms/Mathematical/eluer-totient.cpp (hunk 2 above)
Comment (yashagarwal1999) [reply Algorithms/Mathematical/eluer-totient.cpp]: Added Sample input and output

File: Algorithms/Mathematical/catalan_numbers.cpp (hunk 5 above)
Comment (yashagarwal1999) [reply Algorithms/Mathematical/catalan_numbers.cpp]: I did not understand what you are trying to say.

Comment (Bhavnaharitsa) [reply Algorithms/Mathematical/catalan_numbers.cpp]: No single letter variables

**Comment by yashagarwal1999** [comment]
@Bhavnaharitsa I have made the required changes

**Review by chanmol1999 — APPROVED** [review]

**Pull request merged**